### PR TITLE
Eval cleanups - timing and env.step() 

### PR DESCRIPTION
**Pull request by kantneel**
# Refactor Environment Timing and Pause/Unpause System

This PR implements a comprehensive refactor of the Factorio Learning Environment's timing system, removes the problematic value accrual mechanism, and adds robust pause/unpause functionality.

### Removed Value Accrual Time from Environment
- **Eliminated `value_accrual_time` parameter** from `FactorioGymEnv.__init__()`
- **Removed artificial sleep delays** that were causing timing issues and making episodes unnecessarily slow
- **Reordered `env.step()` logic** to calculate rewards immediately after action execution rather than after waiting
- Actions now execute and return rewards instantly, dramatically improving episode speed

### Enhanced Instance Pause/Unpause System
- **Added comprehensive pause/unpause functionality** to `FactorioInstance`
- **Separated speed and pause controls** - speed settings are preserved when pausing/unpausing
- **Added convenience methods**:
  - `set_speed_and_unpause()` - common use case for starting actions
  - `pause_at_speed()` - set speed but pause immediately
  - `toggle_pause()` - switch between paused/unpaused states
  - `is_paused()` - check current pause state
- **Added `pause_after_action` parameter** to `FactorioGymEnv` - enables pausing between trajectory steps
- **Visual feedback** - pause/unpause actions now display timestamped messages in-game

### Server Admin Configuration
- **Added `server-adminlist.json`** to enable `/editor` mode and other admin commands
- Includes `"client_master"` as admin to allow full server control during development/testing

### Comprehensive Test Suite Additions
- **Added `test_elapsed_ticks.py`** with 15+ new tests covering:
  - Sleep timing at different game speeds
  - Move, craft, and harvest timing validation  
  - Real-world sleep duration accuracy
  - Tick accumulation across multiple actions
  - Edge cases like very low speeds
- **Enhanced existing tests** with proper pause/unpause handling
- **Added game_info observation tests** to verify tick tracking in observations

### Achievements System Cleanup
- **Simplified achievements calculation** by removing `AchievementTracker` class
- **Flattened achievement logic** into a single `calculate_achievements()` function
- **Removed achievements from task metadata** to reduce complexity
- More maintainable and straightforward achievement processing

### Tool Timing Improvements
- **Updated all action tools** (`move_to`, `craft_item`, `harvest_resource`, `connect_entities`) to:
  - Track elapsed ticks accurately
  - Sleep for appropriate real-world time based on game speed
  - Provide timing feedback via console output
- **Redesigned `sleep()` tool** to work with standard tick calculations rather than polling
- **Fixed movement calculations** to use consistent character speed (0.15 tiles/tick)

### API Improvements
- **Made `num_agents` optional** in task configurations (defaults to 1)
- **Added command-line argument parsing** to `run_eval.py`
- **Updated observation formatting** to include game timing information

## Migration Notes
- `value_accrual_time` parameter removed from `FactorioGymEnv` - remove from instantiation calls
- Use `set_speed_and_unpause()` instead of separate `set_speed()` calls in most cases
- Tests may need `pause_after_action=False` for immediate execution scenarios

This refactor significantly improves the environment's usability for both training and evaluation while maintaining full backward compatibility for core functionality.

### CODE REVIEW

**Comment by JackHopkins** [comment]
Nice work!

- Why is there a `set_speed_and_unpause` as well as `set_speed` - it seems that the latter method is sufficient? What is the motivation for 2 methods?
- I'm not sold on the toggle, I think we should rely on `set_speed` (to zero or not) to explicitly control this state instead (or use your new method).
- Are the tests in `test_elapsed_ticks` added to the CI/CD suite? If not, it would be a good idea to include them.

**Comment by hrshtt** [comment]
is it possible to parametrize the new tests?
in the following manner:

`@pytest.mark.parametrize("speed", range(1, 10))  # 10 independent items`

sleeping tests done on a single worker starve the other workers in distributed testing

File: fle/env/gym_env/run_eval.py [hunk 1]
```diff
@@ -132,4 +133,14 @@ async def main(config_path):
 
 if __name__ == "__main__":
     multiprocessing.set_start_method("spawn")
-    asyncio.run(main())
+    parser = argparse.ArgumentParser(
```
Comment (kiankyars) [inline fle/env/gym_env/run_eval.py:136]: I removed this previously because we already have the parsing in fle/run.py either we should remove it from there or remove it from here

**Comment by kiankyars** [comment]
> Nice work!
> 
> * Why is there a `set_speed_and_unpause` as well as `set_speed` - it seems that the latter method is sufficient? What is the motivation for 2 methods?

seems to be a convenience method

**Review by kiankyars — COMMENTED** [review]
I reviewed the API improvements only and the other changes look great for QOL from the surface.
